crates/burn-dataset/src/vision/image_folder.rs

```diff
@@ -148,6 +211,20 @@ fn parse_image_annotation(
                 .map(|name| *classes.get(name).unwrap())
                 .collect(),
         ),
+        AnnotationRaw::SegmentationMask(mask_path) => {
+            let mask_image = image_path_to_vec_pixel_depth(mask_path);
+            // assume that each channel in the mask image is the same and
+            // each pixel in the first channel corresponds to a class.
+            // multi-channel image segmentation is not supported at this time.
+            Annotation::SegmentationMask(SegmentationMask {
+                mask: mask_image
+                    .into_iter()
+                    .enumerate()
+                    .filter(|(i, _)| i % 3 == 0)
+                    .map(|(_, pixel)| pixel)
+                    .collect(),
+            })
```

Comment:
The filtering here will probably not be required given the suggested changes in the previous comment.

Reply:
filtering was removed and put in the `segmentation_mask_to_vec_usize`. fixed in the next commit.